source/scheduler.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <queue>
+
+namespace sim {
+
+class Event;
+
+// Scheduler is implemented as a Singleton class
+// which provides a global access to a single instance
+class Scheduler {
+public:
+    // Static method to get the instance
+    static Scheduler& get_instance() {
+        static Scheduler instance;
+        return instance;
+    }
+
+    void tick();
+    void add(const Event& event);
+
```

Comment:
Add method `peek_time()` to peek the time of event in the head of queue.

```suggestion
int peek_time() const;
```

Reply:
I removed this in favor of introducing a Stop event [e835734](https://github.com/Cloud-Storage-Team/algnet/pull/49/commits/e835734dab8b035fa31c1ed5929c52af03e99c24)